extensions/ql-vscode/src/extension.ts

```diff
@@ -1529,6 +1530,35 @@ async function activateWithInstalledDistribution(
   };
 }
 
+function addUnhandledRejectionListener() {
+  const handler = (error: unknown) => {
+    const message = redactableError(
+      asError(error),
+    )`Unhandled error: ${getErrorMessage(error)}`;
+    // Add a catch so that showAndLogExceptionWithTelemetry fails, we avoid
+    // triggering "unhandledRejection" and avoid an infinite loop
+    showAndLogExceptionWithTelemetry(message).catch(
+      (telemetryError: unknown) => {
+        void extLogger.log(
```

Comment:
To test this I tried throwing a test exception, and also added an error to `showAndLogExceptionWithTelemetry`. Annoying something weird is going on and the second error seems to be getting lost. For example it prints
```
Failed to send error telemetry: Cannot read properties of undefined (reading 't')
Unhandled error: Cannot read properties of undefined (reading 't')
```
even though the two messages should be different.

I'm therefore not sure whether to keep this or just log one of the messages.